jib-plugins-common/src/main/java/com/google/cloud/tools/jib/plugins/common/UpdateChecker.java

```diff
@@ -123,11 +127,24 @@ void setDisableUpdateCheck(boolean disableUpdateCheck) {
         // Generate config file if it doesn't exist
         ConfigJsonTemplate config = new ConfigJsonTemplate();
         Files.createDirectories(configDir);
-        try (OutputStream outputStream = Files.newOutputStream(configFile)) {
+        Path temp = configDir.resolve(CONFIG_FILENAME + ".tmp");
+        try (OutputStream outputStream = Files.newOutputStream(temp)) {
           JsonTemplateMapper.writeTo(config, outputStream);
+          // Attempts an atomic move first, and falls back to non-atomic if the file system does not
+          // support atomic moves.
+          try {
+            Files.move(
+                temp,
+                configFile,
+                StandardCopyOption.ATOMIC_MOVE,
+                StandardCopyOption.REPLACE_EXISTING);
+          } catch (AtomicMoveNotSupportedException ignored) {
+            Files.move(temp, configFile, StandardCopyOption.REPLACE_EXISTING);
+          }
         } catch (IOException ex) {
           // If attempt to generate new config file failed, delete so we can try again next time
           log.accept(LogEvent.debug("Failed to generate global Jib config; " + ex.getMessage()));
+          Files.deleteIfExists(temp);
           Files.deleteIfExists(configFile);
```

Comment:
There's one problem with this. If `Files.move` fails, the existing and working config file will be deleted. I think we can just remove `Files.deleteIfExists()` if we are moving the temp file rather than directly replacing the original file. No need to touch the exist one ever. And it's not a big deal not to delete the temp file. It will be extremely rare that moving will fail, and if that happens, the presence of a `.tmp` file may give more clues, I guess?

Reply:
hrmm.. so I removed this for the regular config file, but I left in the temp file removal, maybe I should leave that in?